### DISCUSSION TITLE
Update to latest version

### DISCUSSION THREAD

**Discussion by mlinkey** [BlueMystical/Dark-Mode-Forms#72] [Q&A] [answered]
I love this!  I am trying to update to the latest version and I am getting the attached error message.  I am not sure what is different than the older version we were on.  We are trying to update to 1.8.15 of the Darkmode.cs file.  Any assistance would be greatly appreciated.  Seems simple enough to update but sure not working right.
![Screenshot 2024-12-16 171525](https://github.com/user-attachments/assets/08a55341-aa2d-4ba0-ab86-e378aaa05b34)

**Comment by BlueMystical** [comment]
uff that error looks nasty, says a Win32 API call failed, what Windows do you have?
Out of my head im not sure what that API does, but certainly it isnt the one i use to paint it dark colors.

**Comment by mlinkey** [comment]
Running windows 11!  You have step through the code and it happens right after the run for the form in program.cs. Very strange the older version worked and this one does not.   Wondering if something changed as a requirement with the latest version?  Hoping to do my digging tomorrow on the issue. But my main form never loads which is odd. If I comment out the dm = dark mode{} block it will work. With the default style set it does not

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: BlueMystic ***@***.***>
Sent: Monday, December 16, 2024 7:53:55 PM
To: BlueMystical/Dark-Mode-Forms ***@***.***>
Cc: Linkey, Michael ***@***.***>; Author ***@***.***>
Subject: [External] Re: [BlueMystical/Dark-Mode-Forms] Update to latest version (Discussion #72)


uff that error looks nasty, says a Win32 API call failed, what Windows do you have?
Out of my head im not sure what that API does, but certainly it isnt the one i use to paint it dark colors.

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1AX_KCtzG4juqZzWEkueIYKaPQcRGqG1DBd8F5J7lUXm1c7s65jHvDLtDWl2kAmD-aFPcCcTJKNIzBPVTsmJwBvlawuLgIfBKLuDKwgPOlzEj5HEkxZRasn1_PxbXvKPn3l1XhF5HV9jpn61ti3CiFgG-tg2lxaUircjQzrphxDA3PH0N_JIB4w8356VAFe1-UMDSTRdleyYu8y-5oa4TcXkJ04J3gBqRklJVLQqqYa6l2shAd7qvXJs4vO7VUnO4G1DLYDk8vdgpfhLPmsrDEL4l0iA4ncIv9l2QhQiqq3o98PMFbgZTOQPz3Loa1cyrZaFQ9e5IXzUGv89kIq_GfTlNO12KIXjeW_gcJh_r2nM7zBvjRcAy8V5oSJSwuXLNVjIo3fKuiuJlch4iNEjXaxyBPmnmTP3Oy14tNWXW7TU/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Fdiscussions%2F72%23discussioncomment-11587651>, or unsubscribe<https://secure-web.cisco.com/1g5NtULTHupBJ0PrCW-sya-Aa7CVX2V7roO8GLJYHXTzsQXiWTIJP_P88c4V0ZLhtjzEvSaaCC307z02_GNfZEDjeGeUd2ebCaBxRLXDaD3OT1iNGB3PaqAKJMt14vbEMD268ZEjPBotSG4T6E4OBvb61pjnFcPu5heCkYeFZaFRfx5bTWR2aU3f17X7RKaWkOIOBR6hF41kt0_d-3CG3eTVgW3sFUzJIgOY5pj2Z3GgN6RLiP9rfLUCAm7TG3KZlcz8a5zpn0y9rX_vhDKEsvPnkQff7hiBkyaoHlTN1FAe_1X0EmPRJ_FGVosgOkBO19iqeh5vlAiNf8y75iLvdJWtej2B_fBz8wO6_NDGYzzbTt3wXNn85Dq3MhrmIQwdVaBiwOe_SEe0hnFUX7RvrM8-XEnw8xVRTKe8NOxUo6gs/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIPDGCWM27Z6E6P7OP37HGL2F573HAVCNFSM6AAAAABTXGC5UCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJYG43DKMI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>


State of Illinois - CONFIDENTIALITY NOTICE: The information contained in this communication is confidential, may be attorney-client privileged or attorney work product, may constitute inside information or internal deliberative staff communication, and is intended only for the use of the addressee. Unauthorized use, disclosure or copying of this communication or any part thereof is strictly prohibited and may be unlawful. If you have received this communication in error, please notify the sender immediately by return e-mail and destroy this communication and all copies thereof, including all attachments. Receipt by an unintended recipient does not waive attorney-client privilege, attorney work product privilege, or any other exemption from disclosure.

**Comment by BlueMystical** [comment]
odd indeed, please share your code, if you can, so i can replicate the problem.
Thanks.

**Reply by mlinkey** [reply]
So in my frmstart.cs file, I have added the following: 
        private DarkModeCS dm = null;

In frmstart(), I commented all code out but this and still happens.
{
// Set dark mode if option is set
//if (Intimidator.Properties.Settings.Default.DarkMode)
//{
// https://github.com/BlueMystical/Dark-Mode-Forms/releases
// old version saved location: C:\Users\michael.linkey\OneDrive - State of Illinois\Prog\Intimidator\backup10-31-2019
// new version location:  P:\DC3
//_ = new DarkModeCS(this);  // old method to call dark mode
//dm = new DarkModeCS(this);
dm = new DarkModeCS(this)  // New method to call dark mode
{
    ColorMode = DarkModeCS.DisplayMode.SystemDefault
};
}

This is the last line that runs and then get this error.  This is in the program.cs
Application.Run(new frmStart());

Here is the whole main() from the program.cs
static void Main()
{
    // Add the exception handler to email
    ThreadExceptionHandler handler = new ThreadExceptionHandler();

    // Add the code below to check for the kill file.
    // ** Begin killfile
    if (killSwitchFound())
        return;
    //ApplicationKillSwitch = new DateAndTime.Timer();
    ApplicationKillSwitch = new System.Timers.Timer();
    ApplicationKillSwitch.Elapsed += ApplicationKillSwitch_Tick;
    ApplicationKillSwitch.Interval = 5000;
    ApplicationKillSwitch.Enabled = true;
    // ** End killfile

    Application.ThreadException += handler.OnThreadException;

    // Add MIM users to list
    string ulist = Intimidator.Properties.Settings.Default.MIMAdmins;
    List<string> newlist = new List<string>();
    newlist = ulist.Split('|').ToList();
    foreach (var l in newlist)
    {
        _userlist.Add(l);
    }

    Application.EnableVisualStyles();
    Application.SetCompatibleTextRenderingDefault(false);
    Application.Run(new frmStart());
}

**Reply by mlinkey** [reply]
After the application.run(new frmstart()) it goes here in the darkmode.cs file and this is where it blows up.  Not sure if that helps or not.
![image](https://github.com/user-attachments/assets/bc82429c-0c95-46c7-9b60-3dc9b41b798b)

**Comment by kachnitata** [comment]
I have no experience with debugging this type of issue, but I would recommend:

-verify your project's targeting setup

-check whether you are not running out of disk space on the system partition 

See also https://stackoverflow.com/questions/3343724/how-do-i-pinvoke-to-getwindowlongptr-and-setwindowlongptr-on-32-bit-platforms

**Reply by mlinkey** [reply]
Verified that it is using .NET Framework 4.8.  Confirmed plenty of free space on the disk as well.

**Comment by BlueMystical** [comment]
mmmmm think i found a possible cause:
![image](https://github.com/user-attachments/assets/42844237-e6f5-49bc-a189-24daa1df589f)
@mlinkey: is you Win11 x32 or x64 ?

**Comment by BlueMystical** [comment, marked as answer]
Just published a version having a fix for that API call on Win32 system, however my system is x64, so i can not test this, please let me know how it goes.

https://github.com/BlueMystical/Dark-Mode-Forms/releases/tag/v1.8.16

**Comment by mlinkey** [comment]
We are all x64 here.  So my development machine is Windows 11 x64.  If you have it, I can certainly test it out for you there!

From: BlueMystic ***@***.***>
Sent: Tuesday, December 17, 2024 12:34 PM
To: BlueMystical/Dark-Mode-Forms ***@***.***>
Cc: Linkey, Michael ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [BlueMystical/Dark-Mode-Forms] Update to latest version (Discussion #72)


mmmmm think i found a possible cause:
image.png (view on web)<https://secure-web.cisco.com/1Jnn9-W7oZ6RHxe03Z4zRxQ8lRKNdheyMbriMarhxAnX-VwJCoCSeZPSdnFRrq6GlJP7yKbL_amNkLghhYuaFeQLV73-PngFrD0xRmNmE3knuXxQdk1PrL_VQGRYHRumMZa-oRiGtUPCH0VzjqFhecL5dMc-Qp3b470o_g8854rFo2cqAer2ret3MjLbgFfvFCm9EKDZM-zNfMa2llZ65MfyguA9flH6BgNTwawqG4n85_uqK20p7BmV52cVT9zaH7xeP87nXsPfFj_N9EVoP_U1_AKIQ2rvP4nE1PE_1tP0UxzDkenIYVNSsTNvIA6yyQy9j55tcoeKottkYigXxfeB9Rm3P_3Hi18ajExWHTHxuUuYNssU08Yuzs7kuk7GMYnuHfImC8GxmSaW22Z64VvXg6Y4ZNW8VA1pCIazsN-c/https%3A%2F%2Fgithub.com%2Fuser-attachments%2Fassets%2F42844237-e6f5-49bc-a189-24daa1df589f>
@mlinkey<https://secure-web.cisco.com/1bNQ1eXkMILBux2A2cFP4I7nrgJHWCDJnOPsBJZXJMvU6MQQS2IqgCWuOhCC_UZA6M_eOxAwkSbAWQX1hTcfB4OaqHwtj_sQoBUk0UbKkAfe2I2eunfFeE0Il38I3T-E1Nwr66FWpbtGSLpeX9Upyjjiw9f2A_dutt7cB5j5E8-H2rP_D3HmOJ_zWlGiLaATP5ojayweFEcyjblBNuj3mkYC2jVIvCOEaMT_aBfgzo8WYpBqwqQbkNeXYxWk7298Wgje96fXHVSvpwg1gnKJAs4tewVswIngFKzomOM0tmBDNeEyNh9J13CCOtNudh_cBvUtX_DcB7yXCo3CKk0b_A3OSFnSy7tr3ho0qQPLMuxgbIsBv9UqKz6SSBtPJns_eXGRO6R1k0fWv7FBKALaYuE1BVAfq7T8EOGUTUKtS4QU/https%3A%2F%2Fgithub.com%2Fmlinkey>: is you Win11 x32 or x64 ?

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1AGOLjrh6jogJOjt7gdgVKuNOt3UmohANWJqiYAHZmhjcVMMm6sNjL7pRUeRtbldoZPuh76AeIIg9xElWDAFYnqWXuJFuNns_2vWuj8dFvPSdeADwYVhK9MWjClyS34uKTJmymiIDpmfSp-OUrPc5T5Gd02kToDMbX0z3BsSaZ03uC0cCDquq2KUi6UmVIu0H-YklF_rfVYp02GCwHmbZhq6WVluVaZa4PadHgBwenZLdlCOdGptpHpDtiTC9sNJX4V3-Vwn6xpo9YbOX1fgp5Rqnsr3XjzeZo0uQelWKUIGStqceRS1ENlmzUAL1-pBlJpyNWnFyf2emSBB0BoMuhak_xbtsYbqKf9Ku8QKIEp8KiJ7prAoeH8EigAXQ3wewDmWiR1D0Rsp5JO-hTKm-nRrglo40BPeuHLN22bcCEbw/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Fdiscussions%2F72%23discussioncomment-11597051>, or unsubscribe<https://secure-web.cisco.com/1o-rm6xKcqqEUrzt2lJvxkNjSv7ZqhxMoDpfeQWqoWm8cGcvRhV8KcRihYy8F5qa4DtZUFytohPNEx-XNZEajDEHm5CcqGZznbNTq1Ym24uyqqHY3H-lRQu2tz40NoUDVtcB3ABZnotPE8WXNtOUbxwMgd8obSyQyPNQ9xjA4nxq02PGu_cq43A9aeeBmNhO4euLx7x0eftvtbTFnCvDXFG-_-ebrBv4HMshlSoL8Uwm15HNSh6W5nSy_zkxU9r_N0qLvHYY2HmdvuT4eM4E3UrnFGmykd2fPYvoKjiDXEOB2gTwySEOg6SSKkBKeFnRM-sivyVz7WTTAXcMo4zAlqiIacb0BIUfLJ9I1SH8cMF7Zu0y6eDWvA9oCmTs-tL9JEYlrrwAusgi-fn24hRMPzApoCo52P-7Xoo5MzsLMBNw/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIPDGCXJU2HDAJDOSEHAEO32GBVB5AVCNFSM6AAAAABTXGC5UCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZG4YDKMI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>


State of Illinois - CONFIDENTIALITY NOTICE: The information contained in this communication is confidential, may be attorney-client privileged or attorney work product, may constitute inside information or internal deliberative staff communication, and is intended only for the use of the addressee. Unauthorized use, disclosure or copying of this communication or any part thereof is strictly prohibited and may be unlawful. If you have received this communication in error, please notify the sender immediately by return e-mail and destroy this communication and all copies thereof, including all attachments. Receipt by an unintended recipient does not waive attorney-client privilege, attorney work product privilege, or any other exemption from disclosure.

**Comment by mlinkey** [comment]
Heading to a meeting and will test after that and let you know!

From: BlueMystic ***@***.***>
Sent: Tuesday, December 17, 2024 12:50 PM
To: BlueMystical/Dark-Mode-Forms ***@***.***>
Cc: Linkey, Michael ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [BlueMystical/Dark-Mode-Forms] Update to latest version (Discussion #72)


Just published a version having a fix for that API call on Win32 system, however my system is x64, so i can not test this, please let me know how it goes.

https://github.com/BlueMystical/Dark-Mode-Forms/releases/tag/v1.8.16<https://secure-web.cisco.com/1oiyfLjaq5Tb-8kqzUivrH47hgAIYUEgnk49wVzJVDD_FSMUzEUHekqAvRASWdcoJa62UW0_Poud2AXc-f0GdEGyp5ghkZtA-SM3nsj9eAlQ0MJZxJilC85f3u06FV5E9pmaqJ9NSe1Ih-iJOHd4D4y6BlgqBzemPL06mJ4UVHBtiXlEOaqm2iKEQsxaX4WyL8-gjroqRBAhOk2-3KfCIstnUd4Sce4__3nUu1qGQ_dj8W5Upa_keBRB4bJsgfZOcQjaGmwt_8OYiBzC6xsVIhgWjiRQWbfAd8FYQSq6_TWPbwdKR0TzeJhy9Knz6HpE_Rj9beTpq7-XGJa0FcD6O20ilLzZr73g5a9Jqde88EMYXkrpX3fvfEyqK2YFn_Y4RxsUhOLLUci622DfdGkqeZ_XalFecUCnDWeO0YQ7llPc/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Freleases%2Ftag%2Fv1.8.16>

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1c115lrfuHcA5vAnXcCnRafwMwhjar1wJIr0gOsoct8NOzV8VWvKlFepiyMLxGj6Tt8CGbWYNBcTBOCA9lrVokYCZyq120OfKfEjsEoYJbieE54QgQSVq09DvCzVNX5jCQWaMpJEsduHJw_lG3hDhuGTMxzETZWR2S3nuZ4OWloIB3HeqUIzmH8WCwOEhnDiX6X2o-K0y-xN8im8vXUfOqna7SIVt9KiPlGECl5Ib545gnk7GTHZKLwP_tOSFkXwqlBmCZFGqsphGWgOC7fX1Ba1To-hH4pxQkrW-9gsqTq9OyqwWYi031bzNEzvg0Oh--AF1URUc8zQ0NJzLDdmwFIU78nyXaj4hDLhWg-veWvyQg_8bNEDXxo4tGrFKwOhBuWibMUAVTGzFeGsG9Uz4Pjr6yWgmY82odiivBmO7XmU/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Fdiscussions%2F72%23discussioncomment-11597201>, or unsubscribe<https://secure-web.cisco.com/1evL7Wtk9g3km-E7q7haYFbB2f_ClMhx8rmjKOSYMrHh0E_C-TOkd8_lJ_DzDw3G884F81xAXq8zAyMsvFFOvvoeorrKAxPAJNyRpE4Szxw1f-555Pez6LLsph0e3U9Ujwnmb7brNkCdK2ju1HNUjAYfmfvgLb-TgBvD4bQqBYEOOYngT2zbLd66_z9WjDKRB9TWF-A8NwE4EUSQ2K41vXzxTrXPkFfV7TMY2jFIh8j5uSib7sPuH-M4CqRpLGYIPUUJVXIvDnI9f0Eld8IgZ82xRdxKq4hNtoAP5DGSK3OexRFECROE-8xShBt95dIa9uoVdfYQExpM6drfkwXwYTNNto1HnjxnaM3R-r_U-HyTOA-kOvhAUe3Z6dD5w4mhvEpIlZprG6tyoqJ6EYvcbNeQktL7Sabbv9Y0rMHPxCGA/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIPDGCVVWD3BOHDIQWZEAK32GBW47AVCNFSM6AAAAABTXGC5UCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZG4ZDAMI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>


State of Illinois - CONFIDENTIALITY NOTICE: The information contained in this communication is confidential, may be attorney-client privileged or attorney work product, may constitute inside information or internal deliberative staff communication, and is intended only for the use of the addressee. Unauthorized use, disclosure or copying of this communication or any part thereof is strictly prohibited and may be unlawful. If you have received this communication in error, please notify the sender immediately by return e-mail and destroy this communication and all copies thereof, including all attachments. Receipt by an unintended recipient does not waive attorney-client privilege, attorney work product privilege, or any other exemption from disclosure.

**Comment by mlinkey** [comment]
That fixed up the issue!  THANK YOU!  I love this module for Dark Mode!

I think could have not caught this except my code was being compile for x86 and not x64.  So that is why it was hitting on the 32 bit as it was compiled that way.  Will work on changing  that up to x64 here at somepoint!

Again Thank you for finding that issue so quickly and getting it going!

From: BlueMystic ***@***.***>
Sent: Tuesday, December 17, 2024 12:50 PM
To: BlueMystical/Dark-Mode-Forms ***@***.***>
Cc: Linkey, Michael ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [BlueMystical/Dark-Mode-Forms] Update to latest version (Discussion #72)


Just published a version having a fix for that API call on Win32 system, however my system is x64, so i can not test this, please let me know how it goes.

https://github.com/BlueMystical/Dark-Mode-Forms/releases/tag/v1.8.16<https://secure-web.cisco.com/1oiyfLjaq5Tb-8kqzUivrH47hgAIYUEgnk49wVzJVDD_FSMUzEUHekqAvRASWdcoJa62UW0_Poud2AXc-f0GdEGyp5ghkZtA-SM3nsj9eAlQ0MJZxJilC85f3u06FV5E9pmaqJ9NSe1Ih-iJOHd4D4y6BlgqBzemPL06mJ4UVHBtiXlEOaqm2iKEQsxaX4WyL8-gjroqRBAhOk2-3KfCIstnUd4Sce4__3nUu1qGQ_dj8W5Upa_keBRB4bJsgfZOcQjaGmwt_8OYiBzC6xsVIhgWjiRQWbfAd8FYQSq6_TWPbwdKR0TzeJhy9Knz6HpE_Rj9beTpq7-XGJa0FcD6O20ilLzZr73g5a9Jqde88EMYXkrpX3fvfEyqK2YFn_Y4RxsUhOLLUci622DfdGkqeZ_XalFecUCnDWeO0YQ7llPc/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Freleases%2Ftag%2Fv1.8.16>

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1c115lrfuHcA5vAnXcCnRafwMwhjar1wJIr0gOsoct8NOzV8VWvKlFepiyMLxGj6Tt8CGbWYNBcTBOCA9lrVokYCZyq120OfKfEjsEoYJbieE54QgQSVq09DvCzVNX5jCQWaMpJEsduHJw_lG3hDhuGTMxzETZWR2S3nuZ4OWloIB3HeqUIzmH8WCwOEhnDiX6X2o-K0y-xN8im8vXUfOqna7SIVt9KiPlGECl5Ib545gnk7GTHZKLwP_tOSFkXwqlBmCZFGqsphGWgOC7fX1Ba1To-hH4pxQkrW-9gsqTq9OyqwWYi031bzNEzvg0Oh--AF1URUc8zQ0NJzLDdmwFIU78nyXaj4hDLhWg-veWvyQg_8bNEDXxo4tGrFKwOhBuWibMUAVTGzFeGsG9Uz4Pjr6yWgmY82odiivBmO7XmU/https%3A%2F%2Fgithub.com%2FBlueMystical%2FDark-Mode-Forms%2Fdiscussions%2F72%23discussioncomment-11597201>, or unsubscribe<https://secure-web.cisco.com/1evL7Wtk9g3km-E7q7haYFbB2f_ClMhx8rmjKOSYMrHh0E_C-TOkd8_lJ_DzDw3G884F81xAXq8zAyMsvFFOvvoeorrKAxPAJNyRpE4Szxw1f-555Pez6LLsph0e3U9Ujwnmb7brNkCdK2ju1HNUjAYfmfvgLb-TgBvD4bQqBYEOOYngT2zbLd66_z9WjDKRB9TWF-A8NwE4EUSQ2K41vXzxTrXPkFfV7TMY2jFIh8j5uSib7sPuH-M4CqRpLGYIPUUJVXIvDnI9f0Eld8IgZ82xRdxKq4hNtoAP5DGSK3OexRFECROE-8xShBt95dIa9uoVdfYQExpM6drfkwXwYTNNto1HnjxnaM3R-r_U-HyTOA-kOvhAUe3Z6dD5w4mhvEpIlZprG6tyoqJ6EYvcbNeQktL7Sabbv9Y0rMHPxCGA/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIPDGCVVWD3BOHDIQWZEAK32GBW47AVCNFSM6AAAAABTXGC5UCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZG4ZDAMI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>


State of Illinois - CONFIDENTIALITY NOTICE: The information contained in this communication is confidential, may be attorney-client privileged or attorney work product, may constitute inside information or internal deliberative staff communication, and is intended only for the use of the addressee. Unauthorized use, disclosure or copying of this communication or any part thereof is strictly prohibited and may be unlawful. If you have received this communication in error, please notify the sender immediately by return e-mail and destroy this communication and all copies thereof, including all attachments. Receipt by an unintended recipient does not waive attorney-client privilege, attorney work product privilege, or any other exemption from disclosure.

**Comment by BlueMystical** [comment]
Any time,
Regards.
